### PR TITLE
Adding argument to disable cache

### PR DESCRIPTION
**Pull request by eddynaka**

### CODE REVIEW

File: Src/Sarif.PatternMatcher/AnalyzeOptions.cs [hunk 1]
```diff
@@ -41,5 +41,9 @@ public class AnalyzeOptions : AnalyzeOptionsBase
             HelpText = "The maximum file size (in kilobytes) that will be analyzed.",
             Default = 1024)]
         public int FileSizeInKilobytes { get; internal set; }
+
+        [Option(
+            "disable-validation-caching")]
```
Comment (michaelcfanning) [inline Src/Sarif.PatternMatcher/AnalyzeOptions.cs:46]: >disable-validation-caching [](start = 13, length = 26)

disable-dynamic-validation-caching

DisableDynamicValidationCaching

File: Src/Plugins/ValidatorBase.cs [hunk 2]
```diff
@@ -113,6 +119,11 @@ protected ValidatorBase()
             return validationState;
         }
 
+        public static void DisableValidationCaching(bool disableValidationCaching)
```
Comment (michaelcfanning) [inline Src/Plugins/ValidatorBase.cs:122]: >DisableValidationCaching [](start = 27, length = 24)

Just use a static property for this!

File: Src/Sarif.PatternMatcher/ValidatorsCache.cs [hunk 3]
```diff
@@ -277,10 +309,19 @@ public ValidatorsCache(IEnumerable<string> validatorBinaryPaths = null, IFileSys
                             isValidDynamic = null;
                         }
 
-                        ruleToMethodMap[typeName] = new ValidationMethodPair
+                        MethodInfo disableValidationCaching = type.BaseType.GetMethod(
```
Comment (michaelcfanning) [inline Src/Sarif.PatternMatcher/ValidatorsCache.cs:312]: >GetMethod [](start = 76, length = 9)

Let's use a property for this. It's fine to stick with 'ValidationMethods' as a name. You can retrieve the setter for the property and store it there.

**Review by michaelcfanning — APPROVED** [review]
:shipit:

File: Src/Sarif.PatternMatcher/AnalyzeOptions.cs (hunk 1 above)
Comment (eddynaka) [reply Src/Sarif.PatternMatcher/AnalyzeOptions.cs]: renamed

---
In reply to: [592834129](https://github.com/microsoft/sarif-pattern-matcher/pull/288#discussion_r592834129) [](ancestors = 592834129)

File: Src/Plugins/ValidatorBase.cs (hunk 2 above)
Comment (eddynaka) [reply Src/Plugins/ValidatorBase.cs]: changing to a property or field is a problem. We would need the object to change and we don't have it.
I will leave as a method for now and we can try later to change again

---
In reply to: [592834390](https://github.com/microsoft/sarif-pattern-matcher/pull/288#discussion_r592834390) [](ancestors = 592834390)

**Pull request merged**